OpenSim/Simulation/SimbodyEngine/Joint.cpp

```diff
@@ -640,6 +629,26 @@ void Joint::updateFromXMLNode(SimTK::Xml::Element& aNode, int versionNumber)
                 childNameElt->setValue(childFrameName + "_offset");
             }
         }
+
+        // Version 30506 replaced Joint's CoordinateSet with a "coordinates"
+        // list property.
+        if (documentVersion <= 30506) {
+            if (aNode.hasElement("CoordinateSet")) {
+                auto coordSetIter = aNode.element_begin("CoordinateSet");
+                auto coordIter = coordSetIter->getRequiredElement("objects")
```

Comment:
The XML Browser in 3.3 says
```
<WeldJoint>
    ...
    <CoordinateSet>
        <objects />
        <groups />
    </CoordinateSet>
    ...
</WeldJoint>
```
but the CoordinateSet element is not actually serialized when you print a Model containing a WeldJoint. Currently, this PR requires only that, if a CoordinateSet element exists, then it contains an "objects" element; the `while` loop (below) will prevent an exception if the "objects" element is empty. I'll add more checks, but I would be curious to know how a developer is expected to know this since AFAIK none of the serialized WeldJoints in the repo have a CoordinateSet element.